stdlib/buffer.ml

```diff
@@ -194,3 +194,31 @@ let truncate b len =
       invalid_arg "Buffer.truncate"
     else
       b.position <- len
+
+(** {6 Iterators} *)
+
+let to_seq b =
+  let rec aux i () =
+    if i = b.position then Seq.Nil
```

Comment:
Because of `Buffer.{clear, reset, truncate}`, you should probably test for `i >= b.position` instead.